### PR TITLE
unbreak return code pass-through in test.sh

### PR DESCRIPTION
**Pull request by abe-winter**
## What changed
- revert a recent change that hid failures in 1 of our 2 test suite runs
- add conditional skips in some places with go-detect-race
- (incidental) add ability to run test.sh on a specific package
## Why
- we broke this last month when prepping the test suite for go 1.21
- the breakage only affects `-race` mode -- any failures that appeared in coverage runs would have still failed the test suite
## Hold merging
- [x] wait until people have had a chance to fix previously uncaught errors

### CODE REVIEW

File: etc/test.sh [hunk 1]
```diff
@@ -18,12 +19,14 @@ if [[ "$1" == "race" ]]; then
...
 
 # We run analyzetests on every run, pass or fail. We only run analyzecoverage when all tests passed.
-PION_LOG_WARN=webrtc,datachannel,sctp gotestsum --format standard-verbose $LOGFILE -- -tags=no_skip $RACE $COVER ./...
+PION_LOG_WARN=webrtc,datachannel,sctp gotestsum --format standard-verbose $LOGFILE -- -tags=no_skip $RACE $COVER $TEST_TARGET
 SUCCESS=$?
 
 if [[ $RACE != "" ]]; then
 	cat json.log | go run ./etc/analyzetests/main.go
-	exit $?
+	if [ "$?" != "0" ]; then
```
Comment (dgottlieb) [inline etc/test.sh:27]: I think we want to `exit $SUCCESS` here? IIUC, this will return whether `analyzetests/main.go` succeeded.

Comment (abe-winter) [reply etc/test.sh:27]: good point -- do you want analyze failures to fail the suite?

I don't mind ignoring blips, but would like to know if analyze were failing every time. Ultimately more your area than mine, I'm okay either way.

Comment (dgottlieb) [reply etc/test.sh:27]: Good point returned. Failing when either tests or analyzetests failed is probably the right way to go.

**Review by nicksanford — APPROVED** [review]

**Comment by abe-winter** [comment]
looped in data people to look at the last commit here (cef2980a skipping the two affected tests in services/datamanager/builtin/sync_test.go)

if you would rather fix these, I'll revert my skips and wait for your go-ahead

**Pull request merged**